osquery/bro/BrokerManager.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <broker/broker.hh>
+#include <broker/endpoint.hh>
+#include <broker/message_queue.hh>
+
+#include "osquery/bro/QueryManager.h"
+
+#include <osquery/database.h>
+#include <osquery/status.h>
+#include <osquery/system.h>
+
+#include <algorithm>
+#include <iostream>
+#include <list>
+
+namespace osquery {
+
+class BrokerManager {
+ private:
+  BrokerManager();
+
+ public:
+  // Get a singleton instance
+  static BrokerManager* getInstance() {
```

Comment:
@theopolis Quick question here. I used `new` to create a singleton because it will be on the heap. If I would do:
```
static BrokerManager& getInstance() {
    static BrokerManager bm;
    return bm;
  }
```
as for the `Config`, it would be placed on the Stack. I had this Problem when running the Code as an extension. There have been two threads, each having a singleton on their Stack.

This also applies e.g. for the Distributed and Bro, since they start another threat. Wouldnt this be a Problem?